src/utils/iterateFocusable.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Returns an iterator over all of the focusable elements within `container`.
+ * Note: If `container` is itself focusable it will be included in the results.
+ * @param container The container over which to find focusable elements.
+ * @param reverse If true, iterate backwards through focusable elements.
+ */
+export function* iterateFocusableElements(
+  container: Element,
+  reverse = false
+): Generator<HTMLElement, undefined, never> {
+  const walker = document.createTreeWalker(container, NodeFilter.SHOW_ELEMENT, {
+    acceptNode: node =>
+      node instanceof HTMLElement && node.tabIndex === -1 ? NodeFilter.FILTER_SKIP : NodeFilter.FILTER_ACCEPT
```

Comment:
We should also probably skip disabled elements and elements that are visually hidden. The filter I'm using in my PR: 

```

function visible(el: HTMLInputElement) {
  return !el.hidden && (!el.type || el.type !== 'hidden') && (el.offsetWidth > 0 || el.offsetHeight > 0)
}

function focusable(el: HTMLElement) {
  const inputEl = el as HTMLInputElement
  return inputEl.tabIndex >= 0 && !inputEl.disabled && visible(inputEl)
}
```

Reply:
Ahh, ok, so it looked like `tabIndex` doesn't tell the whole story after all. Weird. After some testing, it looks like disabled, hidden, and even elements with certain styles can still have non-negative `tabIndex` while not being tabbable. Unfortunate. I think this is one of those cases where we will have to do a best-effort and get close, at least until there's a primitive for this.